analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpObjectInitializationTracker.cs

```diff
@@ -153,7 +153,7 @@ private bool IsAllowedValue(ExpressionSyntax expression, SemanticModel semanticM
         /// <remarks>
         /// Currently we do not handle the situation with default and named arguments.
         /// </remarks>
-        private bool ObjectCreatedWithAllowedValue(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
+        private bool ObjectCreatedWithAllowedValue(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel, Func<bool> isDefaultConstructorSafe = null)
```

Comment:
Let's make it bool as suggested in other comments.
Let's make it non-optional on all places to force user to think about it and provide explicit value.
```suggestion
        private bool ObjectCreatedWithAllowedValue(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel, bool isDefaultConstructorSafe)
```